src/java/org/apache/cassandra/locator/TokenMetadata.java

```diff
@@ -112,13 +117,21 @@ public TokenMetadata()
     {
         this(SortedBiMultiValMap.<Token, InetAddress>create(null, inetaddressCmp),
              HashBiMap.<InetAddress, UUID>create(),
-             new Topology());
+             new Topology(), false);
     }
 
-    private TokenMetadata(BiMultiValMap<Token, InetAddress> tokenToEndpointMap, BiMap<InetAddress, UUID> endpointsMap, Topology topology)
+    public TokenMetadata(boolean shouldLogTokenChanges)
+    {
```

Comment:
instead of plumbing though a boolean variable `shouldLogTokenChanges`, what about instead adding a jvm flag `palantir_cassandra.log_token_changes`.

Then, in the logging methods, you can just call `Boolean.getBoolean("palantir_cassandra.log_token_changes")`

In general I want to avoid making changes to method specs where possible.

Reply:
`TokenMetadata` internally also create instances of `TokenMetadata`, which calls methods relevant to the logging ([ref](https://github.com/palantir/cassandra/blob/palantir-cassandra-2.2.18/src/java/org/apache/cassandra/locator/TokenMetadata.java#L649). Adding this flag in the constructor ensures that we only get the logs from the singleton `TokenMetadata` that's used by `StorageService`.